web3-react/src/libs/components/CoinbaseOption.tsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react'
+import {
+  getConnection,
+  ConnectionType,
+  tryActivateConnector,
+  tryDeactivateConnector,
+} from '../connections'
+
+export const CoinbaseOption = ({
+  isActive,
+  connectionType,
+  onActivate,
+  onDeactivate,
+}: {
+  isActive: boolean
+  connectionType: ConnectionType | null
+  onActivate: (connectionType: ConnectionType) => void
+  onDeactivate: (connectionType: ConnectionType | null) => void
+}) => {
+  return (
```

Comment:
nit: for code like this, I usually prefer to do something with an early-return. 
Example:
```
if (isActive && connectionType === ConnectionType.NETWORK) {
   return <div />;
}

return <div />;
```
Also, I feel like a switch/case statement would be more suitable for this particular scenario, just to highlight that we did handle all the cases?

Reply:
Theoretically, you could further simplify this code by making `onClick` to be a function `toggleConnector` and doing conditional logic inside. Then, having another conditional inside button to change `Connect` with `Disconnect` (string)

Reply:
Love it, thanks!